### ISSUE TITLE
Beanstalk Changes "succeed" instantly.

### ISSUE THREAD

**Issue by tecnobrat** [hashicorp/terraform#6242]
When you apply changes to a beanstalk environment, they "succeed" as soon as the API call finishes. However, they can take quite awhile to actually complete after that.

I suggest terraform blocks until the beanstalk environment goes back to a stable state before saying that the apply succeeded.  As it is right now, you can have an apply say it was successful, but its not complete yet.

Also causes issues when applying more changes back to back, since the API will return the old state back, so it tries to apply the same changes again (and fails because the beanstalk environment is in an invalid state)

**Comment by tecnobrat** [comment]
@dharrisio any ideas if this is possible at all?

**Comment by dharrisio** [comment]
@tecnobrat Not sure how I missed this one! Need to update my filters :)

This should be fixed with #6342, are you still seeing this issue with the latest master?

**Comment by tecnobrat** [comment]
Oh, so that PR killed two birds? We should probably make sure the CHANGELOG has a corresponding entry :)

I have not tried on master, we run everything on atlas, which can only run official releases.

Thanks sir! :)

**Comment by dharrisio** [comment]
That's correct, although I just noticed a small issue with #6342 where the `wait_for_ready_timeout` won't be used in an environment update. Fix for that will be submitted shortly!

Here's an example of what I see when I update an environment with this change
```
aws_elastic_beanstalk_environment.default: Modifying...
  setting.2676719765.name:      "TEMPLATE" => ""
  setting.2676719765.namespace: "aws:elasticbeanstalk:application:environment" => ""
  setting.2676719765.value:     "3" => ""
  setting.32445750.name:        "" => "TEMPLATE"
  setting.32445750.namespace:   "" => "aws:elasticbeanstalk:application:environment"
  setting.32445750.value:       "" => "4"
aws_elastic_beanstalk_environment.default: Still modifying... (10s elapsed)
aws_elastic_beanstalk_environment.default: Still modifying... (20s elapsed)
aws_elastic_beanstalk_environment.default: Still modifying... (30s elapsed)
aws_elastic_beanstalk_environment.default: Still modifying... (40s elapsed)
aws_elastic_beanstalk_environment.default: Still modifying... (50s elapsed)
aws_elastic_beanstalk_environment.default: Still modifying... (1m0s elapsed)
aws_elastic_beanstalk_environment.default: Still modifying... (1m10s elapsed)
aws_elastic_beanstalk_environment.default: Still modifying... (1m20s elapsed)
aws_elastic_beanstalk_environment.default: Modifications complete
```

So, you can see that it no longer returns immediately.

**Comment by stack72** [comment]
Hi @tecnobrat 

A fix has been merged for this now (#6358) - thanks to @dharrisio for the fix and thank you for pointing out the error

Paul
